crates/cext/src/circuit.rs

```diff
@@ -1151,3 +1152,133 @@ pub unsafe extern "C" fn qk_circuit_delay(
 
     ExitCode::Success
 }
+
+/// @ingroup QkCircuit
+/// Convert a given circuit to a DAG.
+///
+/// The returned DAG needs to be freed with ``qk_dag_free``.
+///
+/// @param circuit A pointer to the circuit from which to create the DAG.
+/// @param qubit_order The order of the qubits in the output DAG. Defaults to the same order
+///     as the input circuit if NULL. If not NULL, it must be of length ``qk_circuit_num_qubits()``.
+/// @param clbit_order The order of the clbits in the output DAG. Defaults to the same order
+///     as the input circuit if NULL. If not NULL, it must be of length ``qk_circuit_num_clbits()``.
+///
+/// @return A pointer to the new DAG.
+///
+/// # Example
+/// ```c
+/// QkCircuit *qc = qk_circuit_new(0, 0);
+/// QkQuantumRegister *qr = qk_quantum_register_new(3, "qr");
+/// qk_circuit_add_quantum_register(qc, qr);
+/// qk_quantum_register_free(qr);
+///
+/// QkDag *dag = qk_circuit_to_dag(qc, (uint32_t[]){2, 0, 1}, NULL);
+///
+/// qk_dag_free(dag);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to a ``QkCircuit``, or if  
+/// ``qubit_order`` or ``clbit_order`` are not of length ``qk_circuit_num_qubits()`` and ``qk_circuit_num_clbits()``,
+/// respectively.  
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_circuit_to_dag(
+    circuit: *const CircuitData,
+    qubit_order: *const u32,
+    clbit_order: *const u32,
+) -> *mut DAGCircuit {
+    let circuit = unsafe { const_ptr_as_ref(circuit) };
```

Comment:
Missing a (meaningless, but principled) safety comment.

Reply:
Fixed in https://github.com/Qiskit/qiskit/pull/15247/commits/812e45e8205d6b0f3bdb0ba54069bf30a9e63d89